massa-bootstrap/src/server_binder.rs

```diff
@@ -201,28 +218,43 @@ impl<D: Duplex + 'static> BootstrapServerBinder<D> {
 
     #[allow(dead_code)]
     /// Read a message sent from the client (not signed). NOT cancel-safe
-    pub async fn next(&mut self) -> Result<BootstrapClientMessage, BootstrapError> {
-        // read prev hash
+    pub fn next_timeout(
+        &mut self,
+        duration: Option<Duration>,
+    ) -> Result<BootstrapClientMessage, BootstrapError> {
```

Comment:
Another hard coded deserializer to remember

Reply:
#3842